### PR TITLE
[School] B1+B2: Backend Issue: entity/repository for schools and GET all, GET by id, POST operations for Schools

### LINKED ISSUE #19: [School] B1 + B2: Backend Issue: entity/repository for schools and GET all, GET by id, POST operations for Schools
- [ ] entity and repository class for schools table
- [ ] POST /api/schools school (does backend validation of fields at least in terms of required/optional)
- [ ] GET /api/schools/all 
- [ ] GET /api/schools?id=ucsb

### PR DESCRIPTION
**Pull request by ZhenyuYu1**
Added entity/repository for schools and GET all, GET by id, POST operations for Schools
Added tests for GET all, GET by id, POST operations for schools

- [ ] entity and repository class for schools table
- [ ] POST /api/schools school (does backend validation of fields at least in terms of required/optional)
- [ ] GET /api/schools/all 
- [ ] GET /api/schools?id=ucsb 

Closes #19

### CODE REVIEW

File: src/main/resources/db/migration/changes/0007_CreateSchoolsTable.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{ "databaseChangeLog": [
+    {
+        "changeSet": {
+          "id": "changeset-0007",
+          "author": "pconrad",
```
Comment (pconrad) [inline src/main/resources/db/migration/changes/0007_CreateSchoolsTable.json:5]: author should really be `ZhenyuYu1` but ok.

**Review by pconrad — DISMISSED** [review]
LGTM

**Review by justinchung712 — DISMISSED** [review]
lgtm

File: src/main/resources/db/migration/changes/0007_CreateSchoolsTable.json (hunk 1 above)
Comment (justinchung712) [reply src/main/resources/db/migration/changes/0007_CreateSchoolsTable.json]: Done

**Review by justinchung712 — APPROVED** [review]
lgtm

**Pull request merged**